components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ReactNode, ReactElement, CSSProperties, Component, cloneElement } from 'react'
+import BreadcrumbItem, { BreadcrumbItemProps } from './BreadcrumbItem'
+
+export interface BreadcrumbProps {
+  /** crumbs of the data to be shown */
+  crumbs?: string[] | ReactNode[];
+  /** separate to be using between crumbs */
+  separator?: string | ReactNode;
+  /** styles specified by the user */
+  style?: CSSProperties;
+  /** Breadcrumb Items in children */
+  children?: ReactNode[],
+  /** className for user to specify their class */
+  className?: string
+}
+
+export default class Breadcrumb extends Component<BreadcrumbProps, any> {
+  static Item: React.FunctionComponent<BreadcrumbItemProps>
+
+  static defaultProps = {
+    separator: '/'
+  }
+
+  renderBreadcrumbs = (): ReactNode => {
+    const { children, className, separator, style } = this.props;
+    const crumbs = React.Children.map(children, (element: ReactNode, index) => {
+      if (!element) {
+        return element
+      }
+      const sep = index === (children && children.length - 1) ? '' : separator
```

Comment:
Been experimenting with this for a while, created a function which inserts separators at appropriate places - 
```js
 insertSeparators = (crumbs: ReactNode[], separator: ReactNode | string) => {
    const updatedCrumbs: ReactNode[] = []
    crumbs.forEach((crumb: ReactNode, index: number) => {
      updatedCrumbs.push(crumb)
      // Insert nodes till last element
      if (index < crumbs.length - 1) {
        updatedCrumbs.push(
          React.createElement("span", { separator: true }, separator)
        )
      }
    })
    return updatedCrumbs
  }
```

Though this is done but we still need separator as a prop to the Item component since it uses it's children to render each Item and with a separator prop we add a separator, for instance even if I create a list of nodes with inserted separators, the Items will still only know the children passed in them for example - 
```
<Breadcrumb.Item>Dashboard</Breadcrumb.Item>
<Breadcrumb.Item>Reports</Breadcrumb.Item>
<Breadcrumb.Item>50467</Breadcrumb.Item>
```

Reply:
Why's that? a separator can't be a standalone react node?